File: APIClothesEcommerceShop/APIClothesEcommerceShop/DTO/Statistics/RevenueStatisticsResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using APIClothesEcommerceShop.DTO.Statistics.Sub;

namespace APIClothesEcommerceShop.DTO.Statistics
{
    public class RevenueStatisticsResponse
    {
        public decimal TotalRevenue { get; set; } // Tổng doanh thu
        public decimal AverageDailyRevenue { get; set; } // Doanh thu trung bình hàng ngày
        public decimal AverageMonthlyRevenue { get; set; } // Doanh thu trung bình hàng tháng
        public decimal HighestRevenue { get; set; } // Doanh thu cao nhất
        public decimal LowestRevenue { get; set; } // Doanh thu thấp nhất
    }
}

File: ECOMMERCESHOPUXUI/EcommerceProject/src/models/dtos/statisticsDtos/revenueStatisticsResponse.js
export default class RevenueStatisticsResponse {
  constructor() {
    this.totalRevenue = 0.0 // Tổng doanh thu
    this.averageDailyRevenue = 0.0 // Doanh thu trung bình hàng ngày
    this.averageMonthlyRevenue = 0.0 // Doanh thu trung bình hàng tháng
    this.highestRevenue = 0.0 // Doanh thu cao nhất
    this.lowestRevenue = 0.0 // Doanh thu thấp nhất
  }

  static fromApiResponse(data) {
    const response = new RevenueStatisticsResponse()
    response.totalRevenue = data.totalRevenue || 0.0
    response.averageDailyRevenue = data.averageDailyRevenue || 0.0
    response.averageMonthlyRevenue = data.averageMonthlyRevenue || 0.0
    response.highestRevenue = data.highestRevenue || 0.0
    response.lowestRevenue = data.lowestRevenue || 0.0

    return response
  }
}
